### ISSUE TITLE
Channel data read incorrectly from uncompressed .dds

### ISSUE THREAD

**Issue by 7ippo** [python-pillow/Pillow#7986]
### What did you do?

[water6_1x4.zip](https://github.com/python-pillow/Pillow/files/14995144/water6_1x4.zip)

This is a texture in A8R8G8B8 DDS format, where most of the pixels have a B channel of zero. 

If you use Image.open and save it to TGA format you will find that the data for the A and B channels has been swapped, which results in subsequent rendering errors.

![Snipaste_2024-04-17_10-44-49](https://github.com/python-pillow/Pillow/assets/18611554/43e0dc67-da0f-478c-9e5f-9adfab032b7b)  
*Check the texture in PVRTextool. The upper part is ARGB dds file and the lower part is RGBA tga file*

After writting some simple tests I found that different version of pillow library lead to different result and Pillow 6.2.2 will read the channel data correctly. You can use the code below and the dds file above to quickly reproduce the problem. 

### What are your OS, Python and Pillow versions?

* OS: Windows
* Python: 3.10.9 / 2.7.18
* Pillow: 10.3.0 / 6.2.2

demo code(different environments use the same demo code):

```python
from PIL import Image
img = Image.open("water6_1x4.dds")
print(img.mode, img.getpixel((0, 0)))
```

output:

```text
# Python 3.10.9 Pillow 10.3.0
RGBA (192, 192, 192, 0)

# Python 2.7.18 Pillow 6.2.2
RGBA (192, 192, 0, 192)
```

### What do I expect?

I want to obtain the correct channel data using Python3 and Pillow 10.3.0(just like Python2 and Pillow 6.2.2) or figure out why this error is occurring. I'm wondering if I'm using it incorrectly.

**Comment by Yay295** [comment]
Pillow 6.2.2 was released more than four years ago. Would you be able to find a newer version of Pillow that works as expected?

I looked through the history of the TGA image plugin since 6.2.2 and didn't notice anything, so it may a change to some other file that caused this.

**Comment by radarhere** [comment]
The demo code that the user provided doesn't use the TGA format, so I think this is just about DDS.

If I manually eyedrop the top left corner in Gimp, I get (192, 192, 192).
If I convert your image to a PNG using ImageMagick and get the top left pixel using Pillow, I get `(192, 192, 192, 0)`

Could you attach a copy of what you think the image should look like, in a common format like PNG?

I suspect PVRTextool only works on Windows. Is that the case? I don't suppose there is another tool that you know of to verify the results?

**Comment by 7ippo** [comment]
Thanks for the reply. Yes It is more likely related to the DDS plugin. And here is the png I got from PVRTextool's `File - Save Image` function and this what I expect.

![water6_1x4](https://github.com/python-pillow/Pillow/assets/18611554/220650dc-d30e-4826-bef6-aece6598a63f)  
*The PNG format file*

This is also what I see from PVRTextool. I'm not sure if PVRTexTool only works on Windows, but I usually use it to view texture data. 

![POPO-20240417-182502](https://github.com/python-pillow/Pillow/assets/18611554/3875c6b1-acdb-41a4-8d4e-6a1c8ca4d6ab)  
*What I see from PVRTextool, including channel's information and raw texel data.*

**Comment by radarhere** [comment]
Thanks for the image.

My conclusion from my previous comment is that Pillow is not incorrect, and nothing needs to be changed here.

Here's some more evidence to consider
- someone saying that PVRTexTool doesn't read A8R8G8B8 correctly - https://www.gamedev.net/forums/topic/692707-directx-texture-tool/5359502#post_5359503
- in the issue that prompted this new behaviour, here's a user saying that Windows Texture Viewer disagreed with the old behaviour that you are after - https://github.com/python-pillow/Pillow/issues/4264#issuecomment-576216627

If you would like Pillow to create your version of the image from A8R8G8B8 images, you could move the channels around only for those images like so
```python
from PIL import Image
with Image.open("water6_1x4.dds") as img:
	if img.tile[-1][-1] == (32, (16711680, 65280, 255, 4278190080)):
		r, g, b, a = img.split()
		img = Image.merge("RGBA", (g, b, a, r))
	img.save("out.png")
```

**Comment by 7ippo** [comment]
Thank you for your patient explanation. After a careful troubleshooting process lasting for almost two days, we have basically confirmed that the issue is caused by a bug or bizarre behavior in PVRTextool related to uncompressed DDS textures. 

We discovered that, interestingly, the rendering error in our game was not caused by Pillow's DDS reading. Instead, it was precisely due to a brief usage of a newer version of the PVRTextool(v5.5.0) during the update of our cacheserver project, for compressing from DDS to ASTC, which led to the issue. This will turn the texture into yellow(different channel data), just like the png file I uploaded above. But the previous version of PVRTextool(v4.20) will not encounter this issue. Pillow won't either. And we confirmed that if the uncompressed DDS textures are preprocessed by using Pillow they will render correctly in the game.

Viewing the uncompressed DDS textures using tools like (Nvidia Texture Tool)[https://developer.nvidia.com/texture-tools-exporter] shows different results compared to PVRTextool.

![Snipaste_2024-04-19_11-40-49](https://github.com/python-pillow/Pillow/assets/18611554/6bf06275-afd3-401c-a7b7-13ddd50a6fd9)

So it seems that there might be an issue with how PVRTextool handles uncompressed DDS textures, causing problems when viewing or compressing using command-line tools.
